### ISSUE TITLE
[RFC] Base HelpDesk

### ISSUE THREAD

**Issue by DarioLodeiros** [OCA/project#391]
Helpdesk will provide the base to manage support tickets. It can be improved with new related modules to manage tickets in website, timesheet, etc.

I’m not sure about the best way to design the model structure, I think that we can choose between 3 main options:

A) New Model ticket with all logic to manage a “special task”… (it’s mean use a o2m or m2m field that allows generate tasks from ticket, etc.)
B) Use delegated inheritance from task… We should hide the linked tasks in project menu.
C) Use the project.task model itself through normal inheritance using a new boolean field in the project model, for example ‘Is helpdesk project’.

I think proposal A option is more generic… we can have different tasks and tickets linked to same project without interfere with the normal project module logic. Set the helpdesk goal on customer issues resolution and service quality… If we need use timesheets or other tasks features in tickets, we can propose use linked tasks to them, and though these control the time, sales, etc.

Proposed Base Features:

- WorkFlow Ticket
- Team Support Manager (notifications, assigned...)
- Project related to the ticket as an option
- Ticket division into tasks  as an option
- Analytic Tickets
- Tickets Types
- Mail integration

Future Features:
- Website Integration with Portal HelpDesk

**Comment by hbrunn** [comment]
a) is what project_issue in earlier versions did, and I'm currently in the process of migrating our helpdesk away from this. We decided to do all helpdesk stuff with tasks, because in the end, you don't want to have to switch between your tasks and your tickets, a ticket *is* a task, so I'd favor c) a lot for a new implementation.

**Comment by DarioLodeiros** [comment]
Hi! Yes! Finally, the idea that we are developing is based in 2 modules, the first is the HelpDesk base, like a new area in Odoo (Its valid to a shop, service company, etc...) the second module allow to link project module with help desk... Its mean create tickets from project alias and access from the project, or Help Desk Team. I think that will be a good job! :)

**Comment by aitorbouzas** [comment]
[This repository has a Helpdesk system](https://github.com/SythilTech/Odoo/tree/11.0/website_support). Maybe it's worth a look even if it differs from your initial implementation idea.
It also has an interesting automatic mail per ticket state system that maybe'll help you with mail integration.

**Comment by JordiBForgeFlow** [comment]
FYI we have migrated the crm_helpdesk from odoo to 11.0 here: https://github.com/OCA/crm/pull/225 and in july last year we proposed in 9.0 https://github.com/OCA/crm/pull/165 and 10.0 https://github.com/OCA/crm/pull/166. No one ever made a single comment to the PR's!!

@DarioLodeiros, we looked at your PR https://github.com/OCA/project/pull/394 and it implements basic features, most of which are already in the PR's we proposed.

Honestly, I believe that we should work on top of each other, unless there's a clear need to set a new direction.

Our proposal therefore is to help to get crm_helpdesk merged in 9.0, 10.0 and 11.0 first, and then lay out an RFC with a proper roadmap.

cc @pedrobaeza

**Comment by pedrobaeza** [comment]
@DarioLodeiros what is the problem with the old `crm_helpdesk` module that leads you to implement a new one?

**Comment by DarioLodeiros** [comment]
I created this issue in the sprintcode of Sevilla as a proposal... at that time I looked for related jobs and I did not find those PR. (I looked in repositories related to projects :zipper_mouth_face: )
I'm agree with @jbeficent about to work on top of each other. A pity we did not notice in Sevilla. 
Let me check OCA/crm#225, but from what I see, I could work in the ticket and projects integration... ¿?
